.github/workflows/test.yml

```diff
@@ -52,10 +53,57 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install system dependencies
+      - name: Restore GDAL runtime cache
+        id: gdal-cache
+        uses: actions/cache@v4
+        with:
+          path: .gdal-root
+          key: gdal-${{ runner.os }}-jammy-gdalbin-v2
+      - name: Build portable GDAL (cache miss)
```

Comment:
This seems like it quickly grew a lot more complex – why do we need to do more than install `gdal-bin` normally?

Reply:
This is still a draft 😉  -- trying things out, it takes too long for the tests to run currently 😇 

Can you look at my open PRs please? 😇